dask_cuda/device_host_file.py

```diff
@@ -186,3 +189,15 @@ def __iter__(self):
     def __delitem__(self, key):
         self.device_keys.discard(key)
         del self.device_buffer[key]
+
+    def get_from_dev_or_host_buffer(self, key):
+        """Get stored value without moving it from host to device memory.
+        If the value isn't in device memory, `DeviceSerialized` is returned.
+        """
+        if key not in self.device_keys or key in self.device_func:
+            return self.__getitem__(key)
+        else:
+            return self.host_buffer[key]
```

Comment:
Could you please clarify what this is suppose to do that is different from `__getitem__`?

Reply:
`__getitem__` will get from the "top" of cache, meaning it will attempt to get from `device_buffer` and if it has been spilled it will retrieve from host/disk and store it back to the GPU and only then return the object.

Reply:
Sorry maybe I'm still confused. Wouldn't this function more simply be just `return self.host_buffer[key]`?

Reply:
I think the purpose of the function is not to retrieve a spilled buffer, but also return the buffer in case it hasn't been spilled, thus the `__getitem__` call which would be equivalent to `return.device_buffer[key]`. However, I just realized that if the buffer has been spilled to host and ultimately to disk, `__getitem__` will make the buffer be retrieved to host and then to device, which is probably not what @madsbk wants. Maybe for the case where the buffer has been spilled to disk there should be a mechanism to retrieve it to the host and return that?

Reply:
More generally I'm wondering if there is a way for us to avoid the addition of this method. It requires changes upstream, which probably won't be accepted ( https://github.com/dask/distributed/pull/3624 ). Trying to use `__getitem__` would be better if possible. Maybe using multiple Zict layers could be another way to get around this need.

Reply:
I see your point @jakirkham . I'm not sure whether there's much that can be done here without chaging upstream other than modifying the default behavior of this class. I guess generally we could opt to change what `__getitem__` does here, but we would have to check carefully whether there would be any drawbacks to that, for one changing it just to do what this function does would require an extra step to move it to device when required, which is probably undesirable.

I particularly think that the spilling functionality is very difficult to get right, it took us some good few weeks with the original implementation, so we should probably discuss upstream what alternatives we have and only then consider changing default behavior here.